Server crash with scarpet

Server exception: https://hastebin.com/ulacahoqob.log
Crash report: [crash-2021-01-14_13.14.21-server.txt](https://github.com/gnembon/fabric-carpet/files/5816249/crash-2021-01-14_13.14.21-server.txt)

*Renamed to .txt because gihub doesn't accept .sc*
Script: [random_tick.txt](https://github.com/gnembon/fabric-carpet/files/5816251/random_tick.txt)

Server version: 20w51a
Mods present:
- fabric-api-0.28.3+1.17.jar
- fabric-carpet-20w51a-1.4.22+v210113.jar
- ViaBackwards-3.3.0-20w51a (1).jar
- ViaFabric [1.16.3] {0.2.16-586.1}.jar
- ViaVersion-3.3.0-20w51a.jar


Steps to reproduce:
- Run _**/script load random_tick**_
- Run _**/random_tick setDelay 60**_
- Run _**/script load random_tick**_ again

Expected result:
 ‎ ‎  ‎ ‎‎ ‎\- Either not load the values back because I messed up or work fine


Actual result:
  ‎ ‎ ‎ ‎ \- **Server closed**

```[ERROR] Encountered an unexpected exception

java.lang.NoSuchMethodError: net.minecraft.class_2507.method_10633(Ljava/io/File;)Lnet/minecraft/class_2487;

at carpet.script.bundled.FileModule.read(FileModule.java:77) ~[fabric-carpet-20w51a-1.4.21+v201216.jar:?]

at carpet.script.bundled.Module.getData(Module.java:38) ~[fabric-carpet-20w51a-1.4.21+v201216.jar:?]

at carpet.script.CarpetScriptHost.loadState(CarpetScriptHost.java:673) ~[fabric-carpet-20w51a-1.4.21+v201216.jar:?]

at carpet.script.CarpetScriptHost.<init>(CarpetScriptHost.java:75) ~[fabric-carpet-20w51a-1.4.21+v201216.jar:?]

at carpet.script.CarpetScriptHost.create(CarpetScriptHost.java:89) ~[fabric-carpet-20w51a-1.4.21+v201216.jar:?]

at carpet.script.CarpetScriptServer.addScriptHost(CarpetScriptServer.java:240) ~[fabric-carpet-20w51a-1.4.21+v201216.jar:?]

at carpet.commands.ScriptCommand.lambda$register$36(ScriptCommand.java:248) ~[fabric-carpet-20w51a-1.4.21+v201216.jar:?]

at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:262) ~[intermediary-server.jar:?]

at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:176) ~[intermediary-server.jar:?]

at net.minecraft.class_2170.method_9249(class_2170.java:249) ~[intermediary-server.jar:?]

at net.minecraft.class_3244.method_14370(class_3244.java:1220) ~[intermediary-server.jar:?]

at net.minecraft.class_3244.method_31286(class_3244.java:1207) ~[intermediary-server.jar:?]

at net.minecraft.class_3244.method_12048(class_3244.java:1186) ~[intermediary-server.jar:?]

at net.minecraft.class_2797.method_12115(class_2797.java:36) ~[intermediary-server.jar:?]

at net.minecraft.class_2797.method_11054(class_2797.java:9) ~[intermediary-server.jar:?]

at net.minecraft.class_2600.method_11072(class_2600.java:21) ~[intermediary-server.jar:?]

at net.minecraft.class_3738.run(class_3738.java:18) ~[intermediary-server.jar:?]

at net.minecraft.class_1255.method_18859(class_1255.java:144) ~[intermediary-server.jar:?]

at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[intermediary-server.jar:?]

at net.minecraft.server.MinecraftServer.method_24306(MinecraftServer.java:760) ~[intermediary-server.jar:?]

at net.minecraft.server.MinecraftServer.method_18859(MinecraftServer.java:157) ~[intermediary-server.jar:?]

at net.minecraft.class_1255.method_16075(class_1255.java:118) ~[intermediary-server.jar:?]

at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:742) ~[intermediary-server.jar:?]

at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:736) ~[intermediary-server.jar:?]

at net.minecraft.class_1255.method_18857(class_1255.java:127) ~[intermediary-server.jar:?]

at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:721) ~[intermediary-server.jar:?]

at net.minecraft.server.MinecraftServer.handler$zem000$modifiedRunLoop(MinecraftServer.java:3317) ~[intermediary-server.jar:?]

at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:650) ~[intermediary-server.jar:?]

at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257) ~[intermediary-server.jar:?]

at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]```

--- a/src/main/java/carpet/script/bundled/FileModule.java
+++ b/src/main/java/carpet/script/bundled/FileModule.java
@@ -76,9 +76,14 @@ public class FileModule extends Module
         }
         catch (IOException e)
         {
-            try
+            // Copy of NbtIo.read(File) because that's now client-side only
+            if (!file.exists())
             {
-                return NbtIo.read(file);
+                return null;
+            }
+            try (DataInputStream in = new DataInputStream(new FileInputStream(file)))
+            {
+                return NbtIo.read(in);
             }
             catch (IOException ioException)
             {

